src/sagemaker/tuner.py

```diff
@@ -318,6 +319,44 @@ def _prepare_for_tuning(self, job_name=None, include_cls_metadata=False):
         """Prepare the tuner instance for tuning (fit)."""
         self._prepare_job_name_for_tuning(job_name=job_name)
         self._prepare_static_hyperparameters_for_tuning(include_cls_metadata=include_cls_metadata)
+        self._prepare_tags_for_tuning()
+
+    def _get_model_uri(
+        self,
+        estimator,
+    ):
+        """Get model uri attribute of ``Estimator`` object.
+
+        This attribute can live in multiple places, and accessing the attribute can
+        raise a TypeError, which needs to be handled.
+        """
+        try:
+            model_data = getattr(estimator, "model_data", None)
+        except TypeError:
+            model_data = None
+        return model_data or getattr(estimator, "model_uri", None)
+
+    def _prepare_tags_for_tuning(self):
+        """Add tags to tuning job (from Estimator and JumpStart tags)."""
+
+        # Add tags from Estimator class
+        estimator = self.estimator or self.estimator_dict[sorted(self.estimator_dict.keys())[0]]
+
+        estimator_tags = getattr(estimator, "tags", []) or []
+
+        if self.tags is None and len(estimator_tags) > 0:
+            self.tags = []
+
+        for tag in estimator_tags:
+            if tag not in self.tags:
+                self.tags.append(tag)
+
+        # Add JumpStart tags
+        self.tags = add_jumpstart_tags(
```

Comment:
question: do you need this here? Wouldn't the estimator have these tags already?

Reply:
Not necessarily.

Reply:
Not necessarily.

Reply:
Could you elaborate please? In which cases would we want the AMT job to be tagged even though the underlying estimator does not have the tags?